internal/explorer/eplorer_utils_test.go

```diff
@@ -0,0 +1,361 @@
+				{
+					Name:        "string_prop",
+					Required:    pointer(true),
+					In:          "path",
+					Description: "hey this is a string, required and overidden!",
+					Schema: base.CreateSchemaProxy(&base.Schema{
+						Type:        []string{"string"},
+						Format:      util.OAS_format_password,
+						Description: "you shouldn't see this because the description is overridden!",
+					}),
+				},
+				{
+					Name:     "bool_prop",
+					Required: pointer(true),
+					In:       "query",
+					Schema: base.CreateSchemaProxy(&base.Schema{
+						Type:        []string{"boolean"},
+						Description: "hey this is a bool, required!",
+					}),
+				},
+				{
+					Name: "float64_prop",
+					In:   "query",
+					Schema: base.CreateSchemaProxy(&base.Schema{
+						Type:        []string{"number"},
+						Format:      "float",
+						Description: "hey this is a float64!",
+					}),
+				},
+			},
+		},
+		"no read op": {
+			noReadOp:   true,
+			readParams: nil,
+			commonParams: []*high.Parameter{
+				{
+					Name:        "string_prop",
+					Required:    pointer(false),
+					In:          "path",
+					Description: "hey this is a string, required and overidden!",
+					Schema: base.CreateSchemaProxy(&base.Schema{
+						Type:        []string{"string"},
+						Format:      util.OAS_format_password,
+						Description: "you shouldn't see this because the description is overridden!",
+					}),
+				},
+			},
+			want: []*high.Parameter{
+				{
+					Name:        "string_prop",
+					Required:    pointer(false),
+					In:          "path",
+					Description: "hey this is a string, required and overidden!",
+					Schema: base.CreateSchemaProxy(&base.Schema{
+						Type:        []string{"string"},
+						Format:      util.OAS_format_password,
+						Description: "you shouldn't see this because the description is overridden!",
+					}),
+				},
+			},
+		},
+	}
+	for name, testCase := range testCases {
+		name, testCase := name, testCase
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			var readOp *high.Operation = nil
+			if !testCase.noReadOp {
+				readOp = &high.Operation{
+					Parameters: testCase.readParams,
+				}
+			}
+
+			resource := Resource{
+				ReadOp:     readOp,
+				Parameters: testCase.commonParams,
+			}
+
+			mergedParameters := resource.ReadOpParameters()
+
+			if diff := cmp.Diff(mergedParameters, testCase.want, cmp.AllowUnexported(base.Schema{}, base.SchemaProxy{}, sync.Mutex{}, high.Parameter{})); diff != "" {
+				t.Errorf("unexpected difference for resource: %s", diff)
+			}
+
+			dataSource := DataSource{
+				ReadOp: &high.Operation{
+					Parameters: testCase.readParams,
+				},
+				Parameters: testCase.commonParams,
+			}
+
+			mergedParameters = dataSource.ReadOpParameters()
+
+			if diff := cmp.Diff(mergedParameters, testCase.want, cmp.AllowUnexported(base.Schema{}, base.SchemaProxy{}, sync.Mutex{}, high.Parameter{})); diff != "" {
+				t.Errorf("unexpected difference for data source: %s", diff)
+			}
```

Comment:
Even though they share the same function internally right now, we'd prefer this single test to be seperated into two different tests:

- `TestReadOpParameters_Resource` -> testing `resource.ReadOpParameters()`
- `TestReadOpParameters_DataSource` -> testing `dataSource.ReadOpParameters()`